### ISSUE TITLE
Excluding alsa-lib prevents lots of packages from being installed

### ISSUE THREAD

**Issue by major** [osbuild/osbuild-composer#921]
Building any images with various X components that require alsa-lib fails:

```
# composer-cli compose start big-fella ami
 Problem 3: conflicting requests
  - package firefox-68.11.0-1.el8_2.x86_64 requires libpipewire-0.2.so.1()(64bit), but none of the providers can be installed
  - package pipewire-libs-0.2.7-1.el8.x86_64 requires libgstvideo-1.0.so.0()(64bit), but none of the providers can be installed
  - package pipewire-libs-0.2.7-1.el8.x86_64 requires libgstaudio-1.0.so.0()(64bit), but none of the providers can be installed
  - package pipewire-libs-0.2.7-1.el8.x86_64 requires libgstallocators-1.0.so.0()(64bit), but none of the providers can be installed
  - package pipewire-0.2.5-1.el8.x86_64 requires libasound.so.2()(64bit), but none of the providers can be installed
  - package pipewire-0.2.5-1.el8.x86_64 requires libasound.so.2(ALSA_0.9)(64bit), but none of the providers can be installed
  - package pipewire-0.2.5-1.el8.x86_64 requires libasound.so.2(ALSA_0.9.0rc4)(64bit), but none of the providers can be installed
  - package pipewire-0.2.5-1.el8.x86_64 requires libasound.so.2(ALSA_0.9.0rc8)(64bit), but none of the providers can be installed
  - package gstreamer1-plugins-base-1.14.0-4.el8.x86_64 requires libasound.so.2()(64bit), but none of the providers can be installed
  - package gstreamer1-plugins-base-1.14.0-4.el8.x86_64 requires libasound.so.2(ALSA_0.9)(64bit), but none of the providers can be installed
  - package gstreamer1-plugins-base-1.14.0-4.el8.x86_64 requires libasound.so.2(ALSA_0.9.0rc4)(64bit), but none of the providers can be installed
  - package gstreamer1-plugins-base-1.14.0-4.el8.x86_64 requires libasound.so.2(ALSA_0.9.0rc8)(64bit), but none of the providers can be installed
  - package gstreamer1-plugins-base-1.16.1-1.el8.x86_64 requires libasound.so.2()(64bit), but none of the providers can be installed
  - package gstreamer1-plugins-base-1.16.1-1.el8.x86_64 requires libasound.so.2(ALSA_0.9)(64bit), but none of the providers can be installed
  - package gstreamer1-plugins-base-1.16.1-1.el8.x86_64 requires libasound.so.2(ALSA_0.9.0rc4)(64bit), but none of the providers can be installed
  - package gstreamer1-plugins-base-1.16.1-1.el8.x86_64 requires libasound.so.2(ALSA_0.9.0rc8)(64bit), but none of the providers can be installed
  - package alsa-lib-1.1.6-3.el8.x86_64 is filtered out by exclude filtering
  - package alsa-lib-1.1.9-4.el8.x86_64 is filtered out by exclude filtering
  - package alsa-lib-1.2.1.2-3.el8.x86_64 is filtered out by exclude filtering
```

But the depsolve succeeds:

```
# composer-cli blueprints depsolve big-fella | grep firefox
    firefox-68.11.0-1.el8_2.x86_64
```

**Comment by bcl** [comment]
This is because when it depsolves the blueprint it doesn't have an output type, so the rpmmd.Depsolve() call doesn't include any excludes. The excludes are part of the distro/output type.

**Comment by ignatenkobrain** [comment]
It would be great to somehow fix it. Essentialy the hardcoding that's done as part of the osbuild-composer is not possible to override externally (to my knowledge) so it is completely impossible to generate images with some packages. E.g. in my case I can't have qcow2 image with `redhat-lsb-core` as it pulls in `nss` transitively which is excluded on a distro/output type.

@bcl or is there some work-around?

**Comment by msehnout** [comment]
Hi Igor, I'm afraid this is still impossible, but definitely something we want to fix (similarly to #1284).
